File: javaSpringShip/src/main/java/pl/battleships/javaspringship/config/GameProperties.java
package pl.battleships.javaspringship.config;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@ConfigurationProperties(prefix = "game")
@Getter
@Setter
public class GameProperties {
    private String team;
    private String opponentUrl;
    private int timeout;
    private long sleep;
}

File: kotlinSpringShip/src/main/kotlin/pl/battleships/kotlinspringship/config/GameProperties.kt
package pl.battleships.kotlinspringship.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConfigurationProperties(prefix = "game")
@ConstructorBinding
data class GameProperties(
    val timeout: Int,
    val sleep: Long = 150,
    val opponentUrl: String,
)
